dcs/task.py

```diff
@@ -1588,12 +1588,459 @@ def __init__(self, value: Values = Values.UseWhenFlyingInSAMWEZ):
+    zInverse: int
 
-    def __init__(self, value=None):
-        super(OptFormation, self).__init__(value)
+    class Values(IntEnum):
+        """Values are for Rotary or Fixed Wing only.  Rotary values are preceded by Rotary."""
+        # RotaryWedge has no VariantIndex or zInverse field.
+        RotaryWedge = 8
+        RotaryEchelonRightSmall = 589825
+        RotaryEchelonRightMedium = 583826
+        RotaryEchelonRightLarge = 589827
+        RotaryEchelonLeftSmall = 590081
+        RotaryEchelonLeftMedium = 590082
+        RotaryEchelonLeftLarge = 590083
+        RotaryFrontRightSmall = 655361
+        RotaryFrontRightMedium = 655362
+        RotaryFrontLeftSmall = 655617
+        RotaryFrontLeftMedium = 655618
+        # RotaryColumn has no VariantIndex or zInverse field.
+        RotaryColumn = 720896
+
+        # Aircraft Values below
+        LineAbreastClose = 65537
+        LineAbreastOpen = 65538
+        LineAbreastGroupClose = 65539
+        TrailClose = 131073
+        TrailOpen = 131074
+        TrailGroupClose = 131075
+        WedgeClose = 196609
+        WedgeOpen = 196610
+        WedgeGroupClose = 196611
+        EchelonRightClose = 262145
+        EchelonRightOpen = 262146
+        EchelonRightGroupClose = 262147
+        EchelonLeftClose = 327681
+        EchelonLeftOpen = 327682
+        EchelonLeftGroupClose = 327683
+        FingerFourClose = 393217
+        FingerFourOpen = 393218
+        FingerFourGroupClose = 393219
+        SpreadFourClose = 458753
+        SpreadFourOpen = 458754
+        SpreadFourGroupClose = 458755
+        WW2_BomberElementClose = 786433
+        WW2_BomberElementOpen = 786434
+        # WW2: Bomber Height Seperation has no VariantIndex field.
+        WW2_BomberElementHeightSeperation = 851968
+        WW2_FighterVicClose = 917505
+        WW2_FighterVicOpen = 917506
+
+    class FormationIndex(IntEnum):
+        """Formations are for Aircraft only (not helicopters) unless explicitly stated otherwise."""
+        LineAbreast = 1
+        Trail = 2
+        Wedge = 3
+        EchelonRight = 4
+        EchelonLeft = 5
+        FingerFour = 6
+        SpreadFour = 7
+        RotaryWedge = 8
+        RotaryEchelon = 9
+        RotaryFront = 10
+        RotaryColumn = 11
+        WW2_BomberElement = 12
+        WW2_BomberElementHeightSeperation = 13
+        WW2_FighterVic = 14
+
+    class VariantIndex(IntEnum):
+        """The distance that the formation is spread apart is dependent on the formation index.
+        Ignore text for rotaries, its always small-medium-large."""
+        Close = 1
+        Open = 2
+        GroupClose = 3
+
+    class ZInverse(IntEnum):
+        """Only used by rotaries.  Combines with FormationIndex to vary left/right."""
+        Right = 0
+        Left = 1
+
+    def __init__(
+        self,
+        value: Values,
+        formationIndex: FormationIndex,
+        variantIndex: Optional[VariantIndex],
+        zInverse: Optional[ZInverse]
```

Comment:
yes, these params should have defaults.
Because the task will be created with a simple `cls()` call, and afterwards just their `params` will be overwritten

Reply:
Thanks!  I have added defaults for an Echelon Right Open formation.